--- libiodbc-3.52.16/iodbcinst/unicode.c~	2023-05-02 13:20:08.000000000 +0200
+++ libiodbc-3.52.16/iodbcinst/unicode.c	2026-06-06 14:49:37.015149483 +0200
@@ -1562,7 +1562,7 @@
 {
   size_t n = 0;
   wchar_t wc;
-  ucs4_t *us = dest;
+  wchar_t *us = dest;
   int count = 0;
   mbstate_t st;
   
--- libiodbc-3.52.16/iodbcinst/unicode.h.orig	2026-06-06 14:40:22.000000000 +0200
+++ libiodbc-3.52.16/iodbcinst/unicode.h	2026-06-06 14:46:41.414385342 +0200
@@ -86,9 +86,11 @@
 #include <wchar.h>
 #endif
 
-typedef unsigned char  utf8_t;
-typedef unsigned short ucs2_t;
-typedef unsigned int   ucs4_t;
+#include <stdint.h>
+
+typedef uint8_t utf8_t;
+typedef uint16_t ucs2_t;
+typedef uint32_t ucs4_t;
 
 typedef enum
   {
